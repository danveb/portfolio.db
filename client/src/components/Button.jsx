import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";

export default function Button({ title, url, rel=null, target=null }) {
  return (
    <div className="button__container">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}