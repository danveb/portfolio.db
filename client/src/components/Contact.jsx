import "../styles/Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="contact__wrapper">
                <div className="contact__main">
                    <div className="contact__title">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact__intro">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                        <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact