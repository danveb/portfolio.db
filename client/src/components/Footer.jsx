import { Link } from "react-router-dom";
import { socials } from "../constants/socials";
import "../styles/Footer.scss"; 

export default function Footer() {
    return (
        <div className="footer">
            <div className="footer__wrapper">
                <div className="footer__main">
                    {/* <div className="footer__head">
                        <h1>Social Links<span>.</span></h1>
                    </div> */}
                    <div className="footer__copyright">
                        <p>© All rights reserved – Danny Bae</p>
                    </div>
                    <div className="footer__social__icons">
                        <ul>
                            {socials.map((social) => (
                                <li key={social.id}>
                                    <Link to={social.url} target={social.target} rel={social.rel}>
                                        <img src={social.img} alt={social.alt} />
                                    </Link>
                                </li>
                            ))}
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    )
}
