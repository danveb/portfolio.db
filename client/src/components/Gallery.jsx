import asador from "../assets/gallery/asador_blank.png"; 
import bobaLife from "../assets/gallery/bobaLife_blank.png"; 
import sailly from "../assets/gallery/sailly_blank.png"; 
import ioniq5 from "../assets/gallery/ioniq5_blank.png"; 
import coini$t from "../assets/gallery/coini$t_blank.png"; 
import "../styles/Gallery.css"; 

export default function Gallery() {
  return (
    <div className="gallery">
      <div className="gallery__wrapper">
        <div className="gallery__card">
          <img src={sailly} alt="" />
          <img src={asador} alt="" />
          <img src={bobaLife} alt="" />
          <img src={ioniq5} alt="" />
          <img src={coini$t} alt="" />
        </div>
      </div>
    </div>
  )
}