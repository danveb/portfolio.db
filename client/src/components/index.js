export { default as Navbar } from "./Navbar"; 
export { default as Menu } from "./Menu"; 
export { default as Gallery } from "./Gallery"; 
export { default as Contact } from "./Contact"; 
export { default as Footer } from "./Footer"; 
export { default as ProjectList } from "./ProjectList"; 
export { default as ProjectDetail } from "./ProjectDetail"; 