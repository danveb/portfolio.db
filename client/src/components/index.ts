export { default as Navbar } from "./Navbar"; 
export { default as Menu } from "./Menu"; 
export { default as Button } from "./Button"; 
export { default as Gallery } from "./Gallery"; 
export { default as Footer } from "./Footer"; 
export { default as ProjectList } from "./ProjectList"; 
export { default as ProjectDetail } from "./ProjectDetail"; 
export { default as ScrollToTop } from "./ScrollToTop"; 
export { default as MapboxGL } from "./MapboxGL"; 
export { default as Note } from "./Note"; 
export { default as MessageSent } from "./MessageSent"; 