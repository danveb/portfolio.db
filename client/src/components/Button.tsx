import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";
import { ButtonProps } from "../types";

export default function Button({ title, url, rel, target }: ButtonProps) {
  return (
    <div className="button__container" data-testid="button">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}