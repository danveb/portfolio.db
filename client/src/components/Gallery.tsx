import { projects } from "../constants/projects";
import "../styles/Gallery.css";

export default function Gallery() {
  return (
    <div className="gallery">
      <div className="gallery__wrapper">
        <div className="gallery__card">
          {projects.map((project) => (
            <img src={project.img} alt={project.alt} />
          ))}
        </div>
      </div>
    </div>
  )
}