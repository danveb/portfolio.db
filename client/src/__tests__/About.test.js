import { render, screen } from "@testing-library/react"; 
import { About } from "../components"; 

describe("About component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<About />); 
    }); 

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<About />); 
        expect(asFragment()).toMatchSnapshot(); 
    }); 

    // getByText 
    test("should display correct title", () => {
        render(<About />);
        const text = screen.getByText("My Story", { exact: true }); 
        expect(text).toBeInTheDocument(); 
    });

    // getByAltText 
    test("should display html logo alt", () => {
        render(<About />); 
        const altText = screen.getByAltText("html logo"); 
        expect(altText).toBeInTheDocument(); 
    }); 
});