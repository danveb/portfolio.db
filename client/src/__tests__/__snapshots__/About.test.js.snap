// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about"
    id="about"
  >
    <div
      class="about__wrapper"
    >
      <div
        class="about__main"
      >
        <div
          class="about__title"
        >
          <h3>
            My Story
            <span>
              .
            </span>
          </h3>
        </div>
        <div
          class="about__img"
        >
          <img
            alt="profile pic bw"
            src="profile-bw.png"
          />
        </div>
        <div
          class="about__story"
        >
          <p>
            My journey started in 2020 when I wrote my first line of code, 
            <span>
              Hello World
            </span>
             on an HTML document. This simple, yet powerful line of code was the catalyst to learning web development (courses on HTML, CSS & JavaScript) and thus building my first static website.
          </p>
          <p>
            Fast-forward to today, most of my web development work is with React. I design and develop web apps that provide a rich user interface/experience. I focus on creating clean, functional, scalable, and testable apps.
          </p>
          <p>
            On weekends I spend time tufting cool rugs, crafting home-made boba tea with friends and dreaming about soccer.
          </p>
        </div>
        <div
          class="about__skills"
        >
          <ul>
            <li>
              <img
                alt="html logo"
                src="html5.png"
              />
            </li>
            <li>
              <img
                alt="css logo"
                src="css.png"
              />
            </li>
            <li>
              <img
                alt="javascript logo"
                src="js.png"
              />
            </li>
            <li>
              <img
                alt="react logo"
                src="react.png"
              />
            </li>
            <li>
              <img
                alt="sass logo"
                src="sass.png"
              />
            </li>
            <li>
              <img
                alt="express logo"
                src="express.png"
              />
            </li>
            <li>
              <img
                alt="psql logo"
                src="psql.png"
              />
            </li>
            <li>
              <img
                alt="mongoDB logo"
                src="mongoDB.png"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
